app/controllers/voucher_adjustments_controller.rb

```diff
@@ -47,6 +47,10 @@ def add_voucher
       return false
     end
 
+    # Clear payments and payment_fee, to not affect voucher adjustment calculation
+    @order.all_adjustments.payment_fee.destroy_all
+    @order.payments.clear
```

Comment:
Sorry but I have to ask, is this potentially clearing records of actual payments? 
I'm guessing at this point the payments (and indeed order) are in an "incomplete" state, so it's ok to delete. But perhaps we should add a scope just in case? Eg:

```
    @order.payments.incomplete.clear
```

I looked at the Payment model and expected to see a `before_destroy` for this but didn't. Maybe it should be there..
@mkllnk do you have any insights?